plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-orders-controller.php

```diff
@@ -25,6 +27,65 @@ class WC_REST_Orders_Controller extends WC_REST_Orders_V2_Controller {
 	 */
 	protected $namespace = 'wc/v3';
 
+	/**
+	 * Meta keys that are not really considered metadata and thus shouldn't appear in the response.
+	 *
+	 * @var array
+	 */
+	protected $hidden_meta_keys = array(
```

Comment:
you can use `is_protected_meta( $meta_key, 'order' )` function for this

Reply:
I'm not sure if this is exactly the same. The metadata I'm trying not to include in the result is that marked as internal by the [CPT datastore](https://github.com/woocommerce/woocommerce/blob/dd794e8e26792d1b466a1f0c6bc5a83b6e0678db/plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php#L19-L77), which does start with "_" (basically what `is_protected_meta()` checks) but is more specific than that.

This metadata is automatically filtered out when the CPT datastore is in use, during the `read_meta()` call (see [(*)](https://github.com/woocommerce/woocommerce/blob/dd794e8e26792d1b466a1f0c6bc5a83b6e0678db/plugins/woocommerce/includes/data-stores/class-wc-data-store-wp.php#L97) and [(**)](https://github.com/woocommerce/woocommerce/blob/dd794e8e26792d1b466a1f0c6bc5a83b6e0678db/plugins/woocommerce/includes/data-stores/class-wc-data-store-wp.php#L112
)) but we can't do that in the COT datastore right now because we're relying on metadata to sync certain properties which haven't been formalized yet (something we've discussed before and it's in the blockers before release) and filtering that out during `read_meta()` would effectively prevent us from, well... reading it and also updating it (or at least would throw [a doing_it_wrong](https://github.com/woocommerce/woocommerce/blob/dd794e8e26792d1b466a1f0c6bc5a83b6e0678db/plugins/woocommerce/includes/abstracts/abstract-wc-data.php#L314))).

I understand this is not ideal, but it'd give buy us some time to have the props discussion and possibly fix in a different PR.
That said, copy & pasting the whole list of meta keys might be a bit too verbose, so I've changed it to an alternative approach which gets the list directly from the CPT datastore, which maybe looks a bit better.